packaging/init/evcc.service

```diff
@@ -10,9 +10,12 @@ StartLimitIntervalSec=10
 StartLimitBurst=10
 
 [Service]
-ExecStart=/usr/bin/evcc
+ExecStart=/usr/bin/evcc --sqlite /var/lib/evcc/evcc.db
 Restart=always
 RestartSec=10
 
+User=evcc
+Group=evcc
```

Comment:
Good point - this would simply the script. I will give it a spin tomorrow.

Reply:
I have no idea what you‘re talking about. Staying tuned :)

Reply:
I did again look into it (was almost tomorrow...) and would rather not use `DynamicUser`. The directories would be created inside /var/lib/private/evcc and symlinked to /var/lib/evcc which introduces unnecessary complexity. Also it would no make the script much simpler. We still would need to create the directory, we still need to copy the database and we would still need to chown (systemd only chowns the containing files if the owner of the directory is wrong, so this might lead to effects, if the user/directory already exists).

The only thing we could do would be to set `StateDirectory` and use the `STATE_DIRECTORY` env variable in the call, but I don't think this would help very much.

Reply:
Yeah, if you have state to migrate in post-install that gets hairy fast. Most of the issues you describe seem to originate from a migration story.

> if the user/directory already exists

You would need to delete the persistent user.

> we would still need to chown (systemd only chowns the containing files if the owner of the directory is wrong

After removing the persistent user the owner/group would certainly mismatch.

> The directories would be created inside /var/lib/private/evcc and symlinked to /var/lib/evcc which introduces unnecessary complexity.

I think the complexity you see is that it only gets created, when the unit is first started. But as you noticed you can create it yourself and I think that is reasonable given that you want to migrate old state.

I think systemd is a very neat tool to manage a service's lifetime. Find below my personal systemd unit running on NixOS. It goes beyond the scope of this PR, especially by including hardening options, but maybe you'll find it interesting or even useful.

```ini
# /etc/systemd/system/evcc.service
[Unit]
After=network-online.target mosquitto.target mosquitto.service

[Service]
Environment="HOME=/var/lib/evcc"
Environment="LOCALE_ARCHIVE=/nix/store/wqzfxm6fni76h5w9m17q7zkl7h7isk96-glibc-locales-2.35-163/lib/locale/locale-archive"
Environment="PATH=/nix/store/57xv61c5zi8pphjbcwxxjlgc34p61ic9-glibc-2.35-163-bin/bin:/nix/store/a7gvj343m05j2s32xcnwr35v31ynlypr-coreutils-9.1/bin:/nix/store/mydc6f4k2z73xlcz7ilif3v2lcaiqv>
Environment="TZDIR=/nix/store/i16lgq16av602nfyws3ps8dd9yj36dwh-tzdata-2022f/share/zoneinfo"

CapabilityBoundingSet=
DeviceAllow=char-ttyUSB
DevicePolicy=closed
DynamicUser=true
ExecStart=/nix/store/qc0vs3by55v0daf7ii854d56kz3nwsf0-evcc-0.109.2/bin/evcc --config /nix/store/fzbldlqmx5ambi3hmf3jhagb3nvc33qp-evcc.yml 
LockPersonality=true
MemoryDenyWriteExecute=true
PrivateTmp=true
PrivateUsers=true
ProcSubset=pid
ProtectClock=true
ProtectControlGroups=true
ProtectHome=true
ProtectHostname=true
ProtectKernelLogs=true
ProtectKernelModules=true
ProtectKernelTunables=true
ProtectProc=invisible
RestrictAddressFamilies=AF_INET
RestrictAddressFamilies=AF_INET6
RestrictAddressFamilies=AF_UNIX
RestrictNamespaces=true
RestrictRealtime=true
StateDirectory=evcc
SystemCallArchitectures=native
SystemCallFilter=@system-service
SystemCallFilter=~@privileged
UMask=0077
User=evcc
WorkingDirectory=/var/lib/evcc
```

Reply:
> You would need to delete the persistent user.

Since DynamicUser CAN work with preexisting users, this can get fishy rather. If you have a user (log reader or similar) that is part of the group evcc and that group is deleted, you will have unexpected consequences. I would suggest sticking with fixed users. You can always use your really strict (in a positive way, you are correct, I find it one of the most hardened unit files that I have seen so far) unit configuration in an override file.

Reply:
@andig @mweinelt Wie sieht es aus? ich würde es so lassen...